ast2vec/source.py

```diff
@@ -1,27 +1,28 @@
-from bblfsh.github.com.bblfsh.sdk.protocol.generated_pb2 import ParseResponse
-from modelforge import generate_meta
-from modelforge.model import Model, split_strings, merge_strings, write_model
+from modelforge.model import split_strings, merge_strings
 from modelforge.models import register_model
 
-import ast2vec
+from ast2vec.uast import UASTModel
 
 
 @register_model
-class Source(Model):
+class Source(UASTModel):
     """
     Model for source-code storage
     """
     NAME = "source"
 
     def construct(self, filenames, sources, uasts):
-        self._filenames = filenames
+        super(Source, self).construct(filenames=filenames, uasts=uasts)
+        if not len(sources) == len(uasts) == len(filenames):
+            raise ValueError("Length of src_codes({}), uasts({}) and filenames({}) is not equal".
```

Comment:
`raise ValueError("Lengths of src_codes({}), uasts({}) and filenames({}) are not equal".`